lighthouse-core/gather/driver.js

```diff
@@ -1423,13 +1423,14 @@ class Driver {
 
   /**
    * @param {LH.Config.Settings} settings
+   * @param {{height?: number, screenHeight: number?, deviceScaleFactor?: number}} [deviceMetricOverrides]
    * @return {Promise<void>}
    */
-  async beginEmulation(settings) {
+  async beginEmulation(settings, deviceMetricOverrides) {
     if (settings.emulatedFormFactor === 'mobile') {
-      await emulation.enableNexus5X(this);
+      await emulation.enableNexus5X(this, deviceMetricOverrides);
     } else if (settings.emulatedFormFactor === 'desktop') {
-      await emulation.enableDesktop(this);
+      await emulation.enableDesktop(this, deviceMetricOverrides);
```

Comment:
Would it be appropriate to hook the options on the driver itself?

Reply:
What do you mean by that?

Reply:
If the extra param could be removed in favor for having it as a prop on `this`

Reply:
I'm not sure we want to clutter driver with additional properties just for the method here. It's a pretty busy class already 😆